File: FCloud3.Entities/Ai/AiConversation.cs
namespace FCloud3.Entities.Ai
{
    public class AiConversation : IDbModel
    {
        public int Id { get; set; }
        /// <summary>创建者用户Id</summary>
        public int UserId { get; set; }
        /// <summary>关联的AI实例配置Id</summary>
        public int AiInstanceConfigId { get; set; }
        /// <summary>对话标题（用户可编辑，默认取第一条用户消息前20字）</summary>
        public string? Title { get; set; }
        /// <summary>创建时关联的词条Id（可为空，0表示无）</summary>
        public int CurrentWikiItemId { get; set; }
        /// <summary>消息数缓存</summary>
        public int MessageCount { get; set; }

        public int CreatorUserId { get; set; }
        public DateTime Created { get; set; }
        public DateTime Updated { get; set; }
        public bool Deleted { get; set; }

        public const int TitleMaxLength = 64;
    }
}

File: FCloud3.AppFront/FCloud3Front/src/models/ai/aiConversation.ts
export interface AiConversation {
    Id: number
    UserId: number
    AiInstanceConfigId: number
    Title: string | null
    CurrentWikiItemId: number
    ModelName: string | null
    MessageCount: number
    IsPinned: boolean
    IsArchived: boolean
    Created: string
    Updated: string
}

export interface AiMessage {
    Id: number
    ConversationId: number
    Role: AiMessageRole
    Content: string | null
    ToolCalls: string | null
    Order: number
    InputTokenCount: number
    OutputTokenCount: number
    ModelName: string | null
}

export enum AiMessageRole {
    System = 0,
    User = 1,
    Assistant = 2,
    Tool = 3
}

export interface AiToolCallInfo {
    Name: string
    Arguments: string
}

export interface AiChatChunk {
    Text: string
    ToolCalls: AiToolCallInfo[] | null
}
